ci-operator/step-registry/fips-check/optional-operator-images-scan/fips-check-optional-operator-images-scan-commands.sh

```diff
@@ -130,17 +130,22 @@ for package in ${package_list}; do
         csv_test_result="${data_dir}/${package}/$csv"
         echo "##Starting scan for CSV '$csv'" >> $csv_test_result
         # retrieve related images from current CSV
-        image_list=$(cat /tmp/$catalog.json | jq -r '.items[].status.channels[]|select(.currentCSV=="'$csv'")|.currentCSVDesc.relatedImages[]' || true);
+        image_list_string=$(cat /tmp/$catalog.json | jq -r '.items[].status.channels[]|select(.currentCSV=="'$csv'")|.currentCSVDesc.relatedImages[]' || true);
+        read -r -a image_list <<< ${image_list_string}
         if $internal; then
-            image_list=$(echo $image_list | sed -r 's/registry.redhat.io/brew.registry.redhat.io/g' )
-            image_list=$(echo $image_list | sed -r 's/registry.stage.redhat.io/brew.registry.stage.redhat.io/g')
+            len=${#image_list[@]}
+            for ((i=0;i<len;i++)); do
+                if [[ ${image_list[$i]} != *"brew"* ]]; then
+                    image_list[$i]=$(echo ${image_list[$i]} | sed -r 's/registry.redhat.io/brew.registry.redhat.io/g' )
+                    image_list[$i]=$(echo ${image_list[$i]} | sed -r 's/registry.stage.redhat.io/brew.registry.redhat.io/g')
+                 fi
+            done
         fi
         # perform fips scan by using check-payload tool
-        for image_url in ${image_list}; do
+        for image_url in ${image_list[*]}; do
             echo "###test result for image '$image_url' is:" >> $csv_test_result
```

Comment:
@xiaojiey does `read -r -a image_list <<< ${image_list_string}` work well?

Reply:
@xiaojiey if manually executing above:
```
$ catalog="qe-app-registry"
$ oc get packagemanifests.packages.operators.coreos.com -l catalog=$catalog -o json > /tmp/$catalog.json || true
$ package_list=$(cat /tmp/$catalog.json | jq -r '.items[].metadata.name' || true)
$ echo "$package_list"
$ package=openshift-cert-manager-operator
$ currentCSVs=$(cat /tmp/$catalog.json | jq -r '.items[] | select(.metadata.name=="'${package}'") | .status.channels[] | select(.currentCSVDesc.annotations["features.operators.openshift.io/fips-compliant"]=="true") | .currentCSV'| sort | uniq || true);
$ echo "$currentCSVs"
$ csv=cert-manager-operator.v1.15.0
$ image_list_string=$(cat /tmp/$catalog.json | jq -r '.items[].status.channels[]|select(.currentCSV=="'$csv'")|.currentCSVDesc.relatedImages[]' || true);
$ echo "$image_list_string"
registry.redhat.io/xxx/xxx-kube-rbac-proxy@sha256:3bdc...
brew.registry.redhat.io/xxx/xxx-cert-manager-operator-bundle@sha256:b341...
registry.redhat.io/xxx/xxx-operator-rhel9@sha256:281b...
registry.redhat.io/xxx/xxx-manager-acmesolver-rhel9@sha256:1d75...
registry.redhat.io/xxx/xxx-manager-rhel9@sha256:e351...
registry.redhat.io/xxx/xxx-manager-acmesolver-rhel9@sha256:1d75...
registry.redhat.io/xxx/xxx-manager-rhel9@sha256:e351...
registry.redhat.io/xxx/xxx-proxy@sha256:3bdc...
brew.registry.redhat.io/rh-osbs/xxx-cert-manager-operator-bundle@sha256:b341...
registry.redhat.io/cert-manager/xxx-operator-rhel9@sha256:281b...
```
The `read -r -a image_list <<< ${image_list_string}` only generates length 1:
```
$ read -r -a image_list <<< ${image_list_string}
$ echo "${image_list[*]}"
registry.redhat.io/xxx/xxx-kube-rbac-proxy@sha256:3bdc...
$ len=${#image_list[@]}
$ echo $len
1
```

Reply:
@xiaojiey , opening your "Log with this PR", downloading the fips-check-optional-operators-images-scan.tar and untar, check the result:
```
$ grep "test result for image" openshift-cert-manager-operator/cert-manager-operator.v1.15.0 | wc -l
1
```
Length is also only 1.

Reply:
Weird, the `read -r -a image_list <<< ${image_list_string}` works for me locally. I cannot use image_list=($image_list), the shell check will fail. Let me check what is problem upstream.
```
...
#Starting scan for package 'run-once-duration-override-operator'
-e ##image_list before: registry.redhat.io/run-once-duration-override-operator/run-once-duration-override-rhel9-operator@sha256:ce8d28e9a2887eb4a35b61f3ec4e0d761b4b37c01d3ef4decfd4d0a441d83aae registry.redhat.io/run-once-duration-override-operator/run-once-duration-override-rhel9@sha256:1a8a42d686ecdc7febbddda357f10cdef3f1b226712910dee81a84d1be8f2074 brew.registry.redhat.io/rh-osbs/run-once-duration-override-operator-metadata-rhel-9@sha256:78a587df455adeea87d53b63e98743c432e1c4a137764e332887aef4d269430c
the len is 3
##image_list after: brew.registry.redhat.io/run-once-duration-override-operator/run-once-duration-override-rhel9-operator@sha256:ce8d28e9a2887eb4a35b61f3ec4e0d761b4b37c01d3ef4decfd4d0a441d83aae brew.registry.redhat.io/run-once-duration-override-operator/run-once-duration-override-rhel9@sha256:1a8a42d686ecdc7febbddda357f10cdef3f1b226712910dee81a84d1be8f2074 brew.registry.redhat.io/rh-osbs/run-once-duration-override-operator-metadata-rhel-9@sha256:78a587df455adeea87d53b63e98743c432e1c4a137764e332887aef4d269430c
-e
```